### ISSUE TITLE
Give a warning pop-up when opening a .bk2 with a different ROM

### ISSUE THREAD

**Issue by ThunderAxe31** [TASVideos/BizHawk#2328]
When you open a .bk2 movie, BizHawk compares the ROM hash with the one stored in the Header.txt of the .bk2 file. If they don't match, a hud warning message appears for a moment in the lower left: "Warning: Movie hash does not match the ROM". The issue is that this message is very easy to miss, due to its short duration and the presence of other messages at the same time. My request is to have a warning pop-up window for this specific event, so that it can't possibly be missed by the user.

**Comment by RetroEdit** [comment]
Yeah, this should probably be more high-profile. TAS movies are probably the most common use for the movies feature, and accidentally missing that information when trying to playback a TAS movie is annoying (really, missing that information for any movie is probably annoying, since it could potentially affect sync).

**Comment by RetroEdit** [comment]
Unfortunately, this is unlikely to happen soon, because adelikat considers #2336 a blocker for implementing this. The concern here is that in certain cases, the hash warning is wrong, so having a pop-up is considered too hostile to the user.

**Comment by RetroEdit** [comment]
As of 2024, we should elevate the priority of this.

Consider this: we show a full dialog box when CycleCount doesn't match! CycleCount is purely informational, and doesn't affect functionality whatsoever.

By contrast this is actually an important warning, but it's so easy to miss. I missed it just now when trying to play back a movie and it wasted my time. 

If we're concerned about systems where hashes don't match, we can fix the underlying problem, instead of hiding information. If we're *extremely* concerned, we can caveat the message by adding a line afterwards `In certain cases the movie hash may not match the ROM because of how hash calculation currently works.` I'll try and submit a patch myself soon.

**Comment by YoshiRulz** [comment]
It should always go through the Play Movie dialog, problem solved.

**Comment by RetroEdit** [comment]
Thought about Yoshi's comment a bit and about what I want as a user and here's what I think: both drag+drop and `Recent movies` should display an info screen akin to the `Play movie` screen when loaded, but have a checkbox `Show movie info on load` that can be unchecked. This resolves the visibility of the and also encompasses other potential movie issues like a movie made in an old version.

**Comment by Morilli** [comment]
I support making this OSD message into a warning message box and fixing all erroneous hash mismatches.

**Comment by YoshiRulz** [comment]
I stand by my opinion that drag+drop should go via the Play Movie dialog, but that will have to wait for the next release. Morilli already increased the duration of the OSD message in time for 2.10.
